src/line_iterator.rs

```diff
@@ -41,38 +41,40 @@ where
             format: TldrFormat::Undecided,
         }
     }
+}
+
+impl<R: BufRead> Iterator for LineIterator<R> {
+    type Item = LineType;
 
-    pub fn next_token(&mut self) -> Option<LineType> {
+    fn next(&mut self) -> Option<LineType> {
         self.current_line.clear();
         let bytes_read = self.reader.read_line(&mut self.current_line);
         match bytes_read {
             Ok(0) => None,
             Err(e) => {
-                warn!("Could not read line from token reader: {:?}", e);
+                warn!("Could not read line from reader: {:?}", e);
                 None
             }
             Ok(_) => {
                 // Handle new titles
-                if self.first_line && !self.current_line.starts_with('#') {
-                    // It's the new format! Drop next line.
-                    // (Hmm, is there a way to do this without an allocation?)
-                    let mut devnull = String::new();
-                    if let Err(e) = self.reader.read_line(&mut devnull) {
-                        warn!("Could not read line from token reader: {:?}", e);
-                        return None;
-                    }
-                    self.first_line = false;
-                    self.format = TldrFormat::V2;
-                    return Some(LineType::Title(self.current_line.trim_end().to_string()));
-                }
-
                 if self.first_line {
-                    // Clear `first_line` flag
-                    self.first_line = false;
-
-                    // It's the old format.
-                    self.format = TldrFormat::V1;
+                    if self.current_line.starts_with('#') {
+                        // It's the old format.
+                        self.format = TldrFormat::V1;
```

Comment:
Sidenote: I wonder whether we still need to support the old format. The pages are all converted to the new format, right? (If we want to be super-cautious we could emit a deprecation warning when we detect a V1 page, and ask the user to refresh the cache.)

Reply:
Are they? As far as I can tell, none of them are

```
.cache/tealdeer/tldr-master via  v16.3.0 took 5s
❯ fd --type f '.*' pages* | xargs -n1 head -1q |  wc -l
4408
.cache/tealdeer/tldr-master via  v16.3.0 took 6s
❯ fd --type f '.*' pages* | xargs -n1 head -1q | rg '^#' | wc -l
4408
```

or am I misunderstanding something?

Reply:
Ah, you are right! https://github.com/tldr-pages/tldr/pull/958 Then we still need it.